nautobot/core/templates/generic/object_retrieve.html

```diff
@@ -218,21 +207,33 @@ <h1>
                 </div>
             </div>
         {% endif %}
-        {% if object.is_static_group_associable_model and perms.extras.view_staticgroup %}
-            <div id="static_groups" role="tabpanel" class="tab-pane {% if request.GET.tab == 'static_groups' %}active{% else %}fade{% endif %}">
+        {% if object.is_dynamic_group_associable_model and perms.extras.view_dynamicgroup %}
+            <div id="dynamic_groups" role="tabpanel" class="tab-pane {% if request.GET.tab == 'dynamic_groups' %}active{% else %}fade{% endif %}">
+                <div class="row">
+                    <div class="col-md-12">
+                        <div class="alert alert-warning">
+                            Dynamic group membership is cached for performance reasons,
+                            therefore this table may not always be up-to-date.
+                            <br>You can refresh the membership of any specific group by viewing it from the list below
+                            or from the <a href="{% url 'extras:dynamicgroup_list' %}">Dynamic Groups list view</a>.
+                            <br>You can also refresh the membership of all groups by running the
+                            <a href="{% url 'extras:job_run_by_class_path' class_path='nautobot.core.jobs.groups.RefreshDynamicGroupCaches' %}">Refresh Dynamic Group Caches job</a>.
+                        </div>
+                    </div>
+                </div>
                 <div class="row">
                     <div class="col-md-12">
                         <form method="post">
                             {% csrf_token %}
                             <div class="panel panel-default">
                                 <div class="panel-heading">
-                                    <strong>Static Groups</strong>
+                                    <strong>Dynamic Groups</strong>
                                     <div class="pull-right noprint">
                                         <!-- Insert table config button here -->
                                     </div>
                                 </div>
                                 <div class="table-responsive">
-                                    {% render_table associated_static_groups_table 'inc/table.html' %}
+                                    {% render_table associated_dynamic_groups_table 'inc/table.html' %}
```

Comment:
This means that the dedicated dynamic group tab views (e.g. `device_dynamicgroups`) are no longer needed. We should probably at least mark those as deprecated if not remove them now.